From 2e486e80c33fa1a1b0ee62e7dff1d0298d77c7fa Mon Sep 17 00:00:00 2001
From: Stuart Morgan <smorgan@mythtv.org>
Date: Thu, 4 Sep 2014 15:57:06 +0100
Subject: [PATCH 08/11] Fix errors when loading music album art.

Use RemoteFile::exists() instead of FindFile, it's more appropriate
here and faster.

StorageGroup expects a hostname not an IP address, pass the hostname instead.
---
 mythtv/libs/libmythbase/remotefile.cpp           |   15 +++++++++------
 mythtv/libs/libmythmetadata/musicmetadata.cpp    |    8 +++-----
 mythtv/programs/mythbackend/services/content.cpp |    1 +
 3 files changed, 13 insertions(+), 11 deletions(-)

diff --git a/mythtv/libs/libmythbase/remotefile.cpp b/mythtv/libs/libmythbase/remotefile.cpp
index 9ff4b75..fec7e12 100644
--- a/mythtv/libs/libmythbase/remotefile.cpp
+++ b/mythtv/libs/libmythbase/remotefile.cpp
@@ -441,7 +441,7 @@ bool RemoteFile::Exists(const QString &url, struct stat *fileinfo)
 
         if (url.startsWith("myth:"))
         {
-            StorageGroup sGroup(sgroup, host);
+            StorageGroup sGroup(sgroup, gCoreContext->GetHostName());
             fullFilePath = sGroup.FindFile(filename);
             if (!fullFilePath.isEmpty())
                 fileExists = true;
@@ -1173,6 +1173,9 @@ QStringList RemoteFile::FindFileList(const QString& filename, const QString& hos
     // if we are looking for the file on this host just search the local storage group first
     if (gCoreContext->IsThisBackend(hostName))
     {
+        // We could have made it this far with an IP when we really want
+        // a hostname
+        hostName = gCoreContext->GetHostName();
         StorageGroup sgroup(storageGroup, hostName);
 
         if (useRegex)
@@ -1193,17 +1196,17 @@ QStringList RemoteFile::FindFileList(const QString& filename, const QString& hos
             for (int x = 0; x < filteredFiles.size(); x++)
             {
                 strList << gCoreContext->GenMythURL(gCoreContext->GetBackendServerIP(),
-                                                     gCoreContext->GetBackendServerPort(),
-                                                     fi.path() + '/' + filteredFiles[x],
-                                                     storageGroup);
+                                                    gCoreContext->GetBackendServerPort(),
+                                                    fi.path() + '/' + filteredFiles[x],
+                                                    storageGroup);
             }
         }
         else
         {
             if (!sgroup.FindFile(filename).isEmpty())
                 strList << gCoreContext->GenMythURL(gCoreContext->GetBackendServerIP(hostName),
-                                                gCoreContext->GetBackendServerPort(hostName),
-                                                filename, storageGroup);
+                                                    gCoreContext->GetBackendServerPort(hostName),
+                                                    filename, storageGroup);
         }
 
         if (!strList.isEmpty() || !allowFallback)
diff --git a/mythtv/libs/libmythmetadata/musicmetadata.cpp b/mythtv/libs/libmythmetadata/musicmetadata.cpp
index 660ae22..4fc9a62 100644
--- a/mythtv/libs/libmythmetadata/musicmetadata.cpp
+++ b/mythtv/libs/libmythmetadata/musicmetadata.cpp
@@ -790,7 +790,7 @@ QString MusicMetadata::Filename(bool find)
     }
 
     // first check to see if the filename is complete
-    if (QFile::exists(m_filename))
+    if (RemoteFile::Exists(m_filename))
     {
         m_actualFilename = m_filename;
         return m_filename;
@@ -1095,14 +1095,12 @@ QString MusicMetadata::getAlbumArtFile(void)
                 return QString("");
             }
 
-            QString sUrl = RemoteFile::FindFile(url.path(), url.host(), url.userName());
-
-            if (sUrl.isEmpty())
+            if (!RemoteFile::Exists(res))
             {
                 if (albumart_image->embedded)
                 {
                     if (gCoreContext->IsMasterBackend() &&
-                        url.host() == gCoreContext->GetMasterHostName())
+                        url.host() == gCoreContext->GetMasterServerIP())
                     {
                         QStringList paramList;
                         paramList.append(QString("--songid='%1'").arg(ID()));

-- 
1.7.10.2

